happy/Driver.py

```diff
@@ -195,6 +195,28 @@ def __configure(self):
             print "Failed to get state id from %s." % (self.main_conf_file)
             sys.exit(1)
 
+        # default_happy_log_dir will be default dir for happy log
+        # happy_log_environ is os environment variable passed from test-engine
+        try:
+            self.default_happy_log_dir = self.main_conf["default_happy_log_dir"]
+        except Exception:
+            print "Failed to find default_happy_log dir in the main configuration file %s." \
+                % (self.main_conf_file)
+            sys.exit(1)
+
+        try:
```

Comment:
Is there a strong value in adding this additional level of indirection to determine the name of the environment variable to read for the Happy log directory?

Reply:
Yup, by adding this environment variable, we can organize openweave/happy tests logs, we can specify where to save every test device log and happy debug log. Before, everything is under /tmp, no organization for the device logs and happy debug log.

FYI: here is how the environment variable used in openweave test launcher:
https://github.com/openweave/openweave-core/pull/265/files#diff-7dc13b60eed8ccc1a74c2d26886ed8e7

HAPPY_LOG_DIR is similar implementation as HAPPY_STATE_ID for Happy

Reply:
To be clear, I'm not advocated not having HAPPY_LOG_DIR. The question I am raising is whether there is any value in having 'HAPPY_LOG_DIR' as the name of the environment variable to be used to be configurable and loadable via `self.happy_log_environ = self.main_conf["happy_log_environ"]`? I don't think there is and believe the code can and should be just:

```
self.happy_log_environ = "HAPPY_LOG_DIR"
```

Reply:
> To be clear, I'm not advocated not having HAPPY_LOG_DIR. The question I am raising is whether there is any value in having 'HAPPY_LOG_DIR' as the name of the environment variable to be used to be configurable and loadable via `self.happy_log_environ = self.main_conf["happy_log_environ"]`? I don't think there is and believe the code can and should be just:
> 
> ```
> self.happy_log_environ = "HAPPY_LOG_DIR"
> ```

Thanks Grant for your feedback!
I have removed the option from the config file.